app/controllers/RegularContributions.scala

```diff
@@ -70,32 +71,74 @@ class RegularContributions(
       )
     }
 
-  def status(jobId: String): Action[AnyContent] = authenticatedAction().async { implicit request =>
+  def displayFormMaybeAuthenticated(): Action[AnyContent] =
+    maybeAuthenticatedAction(recurringIdentityClientId).async { implicit request =>
+      request.user.fold {
+        val uatMode = testUsers.isTestUserOptionalAuth(request)
+        Future(
+          monthlyContributionsPage(None, uatMode)
+        )
+      } { fUser =>
+        identityService.getUser(fUser).semiflatMap { fullUser =>
+          val uatMode = testUsers.isTestUser(fullUser.publicFields.displayName)
+          Future(monthlyContributionsPage(Some(fullUser), uatMode))
+        } fold (
+          { error =>
+            SafeLogger.error(scrub"Failed to display recurring contributions form for ${fUser.id} due to error from identityService: $error")
+            InternalServerError
+          },
+          identity
+        )
+      }
+    }
+
+  def status(jobId: String): Action[AnyContent] = maybeAuthenticatedAction().async { implicit request =>
     client.status(jobId, request.uuid).fold(
       { error =>
-        SafeLogger.error(scrub"Failed to get status of step function execution for user ${request.user.id} due to $error")
+        SafeLogger.error(scrub"Failed to get status of step function execution for job ${jobId} due to $error")
         InternalServerError
       },
       response => Ok(response.asJson)
     )
   }
 
-  def create: Action[CreateRegularContributorRequest] = authenticatedAction().async(circe.json[CreateRegularContributorRequest]) {
+  def create: Action[CreateRegularContributorRequest] = maybeAuthenticatedAction().async(circe.json[CreateRegularContributorRequest]) {
     implicit request =>
-      SafeLogger.info(s"[${request.uuid}] User ${request.user.id} is attempting to create a new ${request.body.contribution.billingPeriod} contribution")
+      request.user.fold {
+        val result = for {
+          userId <- identityService.getOrCreateUserIdFromEmail(request.body.email)
+          user <- identityService.getUser(IdMinimalUser(userId, None))
+          response <- {
```

Comment:
I don't think we need the extra {} here?

Reply:
👍